items/items.js

```diff
@@ -501,20 +491,11 @@ const removeItem = function (itemOrItemName) {
  * @throws failed to create Item
  */
 const replaceItem = function (itemConfig) {
-  try {
-    const item = getItem(itemConfig.name);
-    if (typeof item !== 'undefined') {
-      removeItem(itemConfig.name);
-    }
-  } catch (e) {
-    if (('' + e).startsWith('org.openhab.core.items.ItemNotFoundException')) {
-      // Item not present
-    } else {
-      throw e;
-    }
+  const item = getItem(itemConfig.name, true);
+  if (item !== null) {
```

Comment:
So it would be good to know if the method actually replaced the item or not.   Maybe the method should at least return the old item or null if it did not exist so that there is some way of knowing if it replaced or inserted?   This is how a java Map `replace()` works actually.

Reply:
FYI I used `replaceItem` by just sticking it into some scripts which needed items to be defined (typically synthetic ones) and it would ensure it was present (or completely fail with an exception) without needing any other error handling. It was more of a 'configuration-as-code' pattern however, which I know you are moving away from.